src/compute-types/src/sinks.rs

```diff
@@ -179,13 +181,48 @@ pub struct CopyToS3OneshotSinkConnection {
     pub prefix: String,
     /// The AWS connection information to do the writes.
     pub aws_connection: mz_storage_types::connections::aws::AwsConnection,
+    /// The ID of the Connection object, used to generate the External ID when
+    /// using AssumeRole with AWS connection.
+    pub connection_id: GlobalId,
+    /// The max file size of each file uploaded to S3.
+    pub max_file_size: u64,
+    /// The relation desc of the data to be uploaded to S3.
+    pub desc: RelationDesc,
+    /// The selected sink format.
+    pub format: CopyFormatParams<'static>,
+}
+
+impl From<CopyToS3OneshotSinkConnection> for CopyToS3Details {
```

Comment:
TODO: I will do some minor refactor to have something like `struct CopyToS3OneshotSinkConnection(CopyToS3Details)` instead of copying all the fields.

Reply:
done